connector/src/main/java/com/linker/connector/network/SocketChannelInitializer.java

```diff
@@ -0,0 +1,56 @@
+package com.linker.connector.network;
+
+import com.linker.connector.configurations.ApplicationConfig;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+import io.netty.handler.codec.LengthFieldPrepender;
+import io.netty.handler.codec.bytes.ByteArrayDecoder;
+import io.netty.handler.codec.bytes.ByteArrayEncoder;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.codec.http.websocketx.WebSocketServerProtocolHandler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SocketChannelInitializer extends ChannelInitializer<SocketChannel> {
+
+    private final static Integer MAX_CONTENT_LENGTH = 8192;
+
+    @Autowired
+    ApplicationContext applicationContext;
+
+    @Autowired
+    ApplicationConfig applicationConfig;
+
+    Long counter = 0L;
+
+    protected void initChannel(SocketChannel ch) throws Exception {
+        ChannelPipeline pipeline = ch.pipeline();
+
+        Integer port = ch.localAddress().getPort();
+        if(port.equals(applicationConfig.getWsPort())){
+            pipeline.addLast(new HttpServerCodec());
+            pipeline.addLast(new HttpObjectAggregator(MAX_CONTENT_LENGTH));
```

Comment:
The lines within conditions can be extracted to different methods I guess.

Reply:
fixed